pkg/agent/provisioning.go

```diff
@@ -537,7 +537,7 @@ func RegisterProvisioner(provisioner provisioning.Provisioning) {
 	if agent.cfg.GetAgentType() == config.DiscoveryAgent || agent.cfg.GetAgentType() == config.GovernanceAgent {
 		agent.proxyResourceHandler.RegisterTargetHandler(
 			"accessrequesthandler",
-			handler.NewAccessRequestHandler(agent.provisioner, agent.cacheManager, agent.apicClient),
```

Comment:
Maybe instead of sending the config in here, we can create a handler in the customunit package. Then that handler is sent in and all the logic in the access request handler will be processed in the new custom unit quota handler?